docs/admin/monitoring.rst

```diff
@@ -165,25 +165,81 @@ NodeStatus can be accessed using the JMX MBean object name
-The ``NodeInfo`` JMX MBean exposes information about the current node;
+The ``NodeInfo`` JMX MXBean exposes information about the current node;
 
-NodeInfo can be accessed using the JMX MBean object name
+NodeInfo can be accessed using the JMX MXBean object name
 ``io.crate.monitoring:type=NodeInfo`` and the following attributes:
 
- - ``ClusterStateVersion``
-
-   Provides the version of the current applied cluster state
-
- - ``NodeId``
-
-   Provides the unique identifier of the node in the cluster
-
- - ``NodeName``
-
-   Provides the human friendly name of the node
++-------------------------+---------------------------------------------------+
+| Name                    | Description                                       |
++=========================+===================================================+
+| ``NodeId``              | Provides the unique identifier of the node in the |
+|                         | cluster.                                          |
++-------------------------+---------------------------------------------------+
+| ``NodeName``            | Provides the human friendly name of the node.     |
++-------------------------+---------------------------------------------------+
+| ``ClusterStateVersion`` | Provides the version of the current applied       |
+|                         | cluster state.                                    |
++-------------------------+---------------------------------------------------+
+| ``ShardStats``          | Statistics about the number of shards located on  |
+|                         | the node.                                         |
++-------------------------+---------------------------------------------------+
+| ``ShardInfo``           | Detailed information about the shards located on  |
+|                         | the node.                                         |
++-------------------------+---------------------------------------------------+
+
+``ShardStats`` returns a `CompositeData`_ object containing statistics about
+the number of shards located on the node with the following attributes:
+
++-------------------+---------------------------------------------------------+
+| Name              | Description                                             |
++===================+=========================================================+
+| ``Total``         | The number of shards located on the node.               |
++-------------------+---------------------------------------------------------+
+| ``Primaries``     | The number of primary shards located on the node.       |
++-------------------+---------------------------------------------------------+
+| ``Replicas``      | The number of replica shards located on the node.       |
++-------------------+---------------------------------------------------------+
+| ``Unassigned``    | The number of unassigned shards in the cluster. If the  |
+|                   | role of the node is ``MASTER`` this will show the total |
```

Comment:
I think only the currently elected master should show the total number of unassigned shards.

If we do this for each eligible master node then we'd have `eligible_master_nodes * unassigned_shards` number of unassigned shards.

*Update*
Looks like the implementation handles it correctly - but the docs here can be a bit misleading?